test/k8sT/Services.go

```diff
@@ -1026,20 +1029,27 @@ var _ = Describe("K8sServicesTest", func() {
 				testCurlFromPodInHostNetNS(httpURL, count, 0, k8s1NodeName)
 				testCurlFromPodInHostNetNS(tftpURL, count, 0, k8s1NodeName)
 			}
-			// In-cluster connectivity from k8s2 to k8s1 IP will still work in
-			// kube-proxy free case since we'll hit the wildcard rule in bpf_sock
+			// In-cluster connectivity from k8s2 to k8s1 IP will still work with
+			// HostRechableServices (regardless of if we are running with or
+			// without kube-proxy) since we'll hit the wildcard rule in bpf_sock
```

Comment:
Could you extend the comment saying why it won't work on kube-proxy?

Reply:
Added the following comment:
```
// This is a known incompatibility with kube-proxy:
// kube-proxy 1.15+ will only load-balance requests from k8s1 to k8s1,
// but not from k8s2 to k8s1. In the k8s2 to k8s1 case, kube-proxy
// would send traffic to k8s1, where it would be subsequently
// dropped, because k8s1 has no service backend.
// However, if HostReachableServices is enabled, then Cilium does
// the service translation already on the client node, bypassing
// kube-proxy completely.
```